src/v/cluster/security_frontend.h

```diff
@@ -35,19 +35,44 @@ class security_frontend final {
       ss::sharded<ss::abort_source>&,
       ss::sharded<security::authorizer>&);
 
+    // Should be called ONLY on the controller leader
     ss::future<std::error_code> create_user(
       security::credential_user,
       security::scram_credential,
       model::timeout_clock::time_point);
 
+    // Should be called ONLY on the controller leader
     ss::future<std::error_code>
       delete_user(security::credential_user, model::timeout_clock::time_point);
 
+    // Should be called ONLY on the controller leader
     ss::future<std::error_code> update_user(
       security::credential_user,
       security::scram_credential,
       model::timeout_clock::time_point);
 
+    // Should be called ONLY on the controller leader
```

Comment:
nit: it would be nice to add proper documentation to this class and these methods.

Also, by leader can we specify this needs to be the leader **node**, this way it's more clear that it can be called from any shard, not just shard0 on the leader.

Reply:
fair points

Reply:
updated existing comments, but I'm going hold off full documentation of these methods for a follow-on PR. I don't want to clutter this diff up too much.